server.js

```diff
@@ -0,0 +1,219 @@
+import {PassThrough} from 'node:stream'
+import {URL, pathToFileURL} from 'node:url'
+
+import {unified} from 'unified'
+import {engine} from 'unified-engine'
+import {VFile} from 'vfile'
+import {
+  Diagnostic,
+  DiagnosticSeverity,
+  Position,
+  Range,
+  TextDocumentSyncKind,
+  TextEdit
+} from 'vscode-languageserver/node.js'
+
+/**
+ * Convert a unist point to a language server protocol position.
+ *
+ * @param {import('unist').Point} point
+ * @returns {Position}
+ */
+function unistPointToLspPosition(point) {
+  return Position.create(point.line - 1, point.column - 1)
+}
+
+/**
+ * @param {import('unist').Point?} point
```

Comment:
```suggestion
 * @param {import('unist').Point|null|undefined} point
```

I find the `?` confusing in TSDoc, also because it’s different with `strictNullChecks` on.
I recommend explicitly listing which values are supported: `Point|null|undefined`, or one of those nullish values?